flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/cleanup/CleanupRetryStrategyFactory.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.dispatcher.cleanup;
+
+import org.apache.flink.configuration.CleanupOptions;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.util.concurrent.ExponentialBackoffRetryStrategy;
+import org.apache.flink.util.concurrent.FixedRetryStrategy;
+import org.apache.flink.util.concurrent.RetryStrategy;
+
+import java.time.Duration;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * {@code CleanupRetryStrategyFactory} is used to create the {@link RetryStrategy} for the {@link
+ * DispatcherResourceCleanerFactory}. It extracts all the necessary information from the passed
+ * {@link Configuration}.
+ *
+ * @see CleanupOptions
+ */
+public enum CleanupRetryStrategyFactory {
+    INSTANCE;
```

Comment:
This is more of a question than a suggestion, but just so that I understand this more: is this a pattern to use an enum with a single pattern to create a factory, rather than a class?

Reply:
Yes, I used it because the functionality is only of static nature.